### DISCUSSION TITLE
Laplace equation - domain dependent

### DISCUSSION THREAD

**Discussion by radlicka** [gridap/Gridap.jl#942] [Q&A]
Hi,
I am working on the calculation of the magnetic lens field in Gridap. Calculated PDE differs in each domain depending on material. Unfortunately, I was not able to duplicate the procedure from tutorial 11. I attached a much simpler example - Laplace equation 
$\nabla(\epsilon(x,y)\nabla \phi) = 0 $
in 2D on a rectangular domain consisting of two subdomains with different permittivity (different functions $\epsilon$). 
[Laplace.zip](https://github.com/gridap/Gridap.jl/files/12648130/Laplace.zip)

* Is it possible to define $a(u,v)$ so that it contains different function $\epsilon$ depending on the element subdomain location? This approach would allow calculating the problem without introducing interfaces.
* Does anybody know how to write the interface conditions: $\phi_1 = \phi_2$ and $\epsilon_1 \phi_{1,n} = \epsilon_2 \phi_{2,n}$?

Thanks,
Tomas

**Comment by JanSuntajs** [comment]
Hello @radlicka;
I hope it is not too late for an answer (or at least an attempt at it) and this is still relevant.

Right now, I am tackling a similar problem (cannot report on the results just yet), and I am planning to handle
it via the Nitsche's method, following [this resource](https://www.diva-portal.org/smash/get/diva2:1591753/FULLTEXT01.pdf). The implementation of the heterogeneity itself is rather straightforward and can be done in a manner similar to the one described in the 
[tutorial on multi-material linear elasticity](https://gridap.github.io/Tutorials/dev/pages/t003_elasticity/#Tutorial-3:-Linear-elasticity-1). However, to properly account for the interface fluxes, one additionally needs to add some terms to the weak form via the above-mentioned Nitsche's method. 

Hope this helps.
Jan
